source/baseObject.py

```diff
@@ -123,6 +123,29 @@ def invalidateCaches(cls):
 		for instance in cls.__instances.keys():
 			instance.invalidateCache()
 
+class GestureCollector(AutoPropertyType):
+	"""A metaclass used for collecting and caching gestures on a ScriptableObject"""
+
+	def __new__(meta, name, bases, dct):
+		gestures = {}
+		cls = super(GestureCollector, meta).__new__(meta, name, bases, dct)
+		for i in reversed(cls.__mro__):
+			try:
+				gestures.update(getattr(i, "_%s__gestures" % i.__name__))
+			except AttributeError:
+				pass
+		for name, script in dct.iteritems():
+			if not name.startswith('script_'):
+				continue
+			scriptName = name[7:]
+			if hasattr(script, 'gesture'):
+				gestures[script.gesture] = scriptName

```

Comment:
While I agree with having both gesture and gestures keyword arguments to the script decorator, I think the actual script functions should just have one "gestures" attribute. That is, the decorator should merge the two. Otherwise, everything that wants to look at a script's gestures has to check both attributes.